experimental/BasisLieHighestWeight/src/MainAlgorithm.jl

```diff
@@ -400,21 +403,21 @@ function add_by_hand(
   end
 
   # identify coordinates that are trivially zero because of the action on the generator
-  zero_coordinates = compute_zero_coordinates(birational_seq, highest_weight)
+  zero_coordinates = compute_zero_coordinates(birational_seq, highest_weight(M))
 
   # calculate new monomials
   for weight_w in weights_with_non_full_weightspace
     dim_weightspace = weightspaces[weight_w]
     add_new_monomials!(
-      L,
+      L(M),
```

Comment:
For consistency, I would like this function to also get the ModuleData `M` instead of some parts of it.